network/network_windows.go

```diff
@@ -185,6 +190,58 @@ func (nm *networkManager) newNetworkImplHnsV1(nwInfo *NetworkInfo, extIf *extern
 	return nw, nil
 }
 
+// add ipv4 and ipv6 routes (if dual stack mode) to windows Node
+func (nm *networkManager) addNewNetRules(nwInfo *NetworkInfo) error {
+	var (
+		err error
+		out string
+	)
+
+	// get interface name of VM adapter
+	ifName := nwInfo.MasterIfName
+	if !strings.Contains(nwInfo.MasterIfName, ifNamePrefix) {
+		ifName = fmt.Sprintf("%s (%s)", ifNamePrefix, nwInfo.MasterIfName)
+	}
+
+	// add ipv4 and ipv6 new net rules to windows node
+	for _, subnet := range nwInfo.Subnets {
+		prefix := subnet.Prefix.String()
+		gateway := subnet.Gateway.String()
+
+		ip, _, er := net.ParseCIDR(prefix)
+		if er != nil {
+			return fmt.Errorf("[net] failed to parse prefix %s", prefix)
+		}
+		if ip.To4() != nil {
+			// netsh interface ipv4 add route $subnetV4 $hostInterfaceAlias "0.0.0.0" metric=270
+			netshV4DefaultRoute := fmt.Sprintf(netRouteCmd, "ipv4", prefix, ifName, ipv4DefaultHop, "270")
+			if out, err = nm.plClient.ExecuteCommand(netshV4DefaultRoute); err != nil {
+				return fmt.Errorf("[net] Adding ipv4 default route failed: %v:%v", out, err)
+			}
+
+			// netsh interface ipv4 add route $subnetV4 $hostInterfaceAlias $gatewayV4 metric=300
+			netshV4GatewayRoute := fmt.Sprintf(netRouteCmd, "ipv4", prefix, ifName, gateway, "300")
+			if out, err = nm.plClient.ExecuteCommand(netshV4GatewayRoute); err != nil {
+				return fmt.Errorf("[net] Adding ipv4 gateway route failed: %v:%v", out, err)
+			}
+		} else {
+			// netsh interface ipv6 add route $subnetV6 $hostInterfaceAlias "::" metric=270
+			netshV6DefaultRoute := fmt.Sprintf(netRouteCmd, "ipv6", prefix, ifName, ipv6DefaultHop, "270")
```

Comment:
why are we using netsh calls? isn't there hnsapi support for this?

Reply:
HNS will support it later on. Currently CNI push these routes to the windows pod as discussed.